config/routes.rb

```diff
@@ -5,4 +5,9 @@
 
   # Defines the root path route ("/")
   # root "articles#index"
+
+  get "/users/:user_id/posts", to: "posts#index"
+  get "/user/:user_id/post/id", to: "posts#show"
+  get "/users", to: "users#index"
+  get "/users/id", to: "users#show"
```

Comment:
[OPTIONAL]
- Alternatively, you could write the routes like this 
```
  resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end
```

Reply:
Thank you so much for suggesting alternative ways of writing routes!